Zend/zend_objects.c

```diff
@@ -201,6 +203,10 @@ ZEND_API void ZEND_FASTCALL zend_objects_clone_members(zend_object *new_object,
 			i_zval_ptr_dtor(dst);
 			ZVAL_COPY_VALUE_PROP(dst, src);
 			zval_add_ref(dst);
+			if (has_clone_method) {
+				Z_PROP_FLAG_P(dst) |= IS_PROP_REINITABLE;
```

Comment:
Should we just add this flag for readonly properties?

Reply:
This behavior is the result of Tyson's comment: https://github.com/php/php-src/pull/9497#discussion_r990563729. Although then I could probably unconditionally remove the flag after the execution of `__clone` (around line 270) too.

Reply:
That's ok with me. Maybe just add a comment that it's intentional.

```php
// Unconditionally add the IS_PROP_REINITABLE flag. It has no influence on non-readonly properties
// but avoids a potential cache miss of the property_info.
```

Reply:
I added a similar comment to the flag itself, so I'll simply add a shorter version of this comment on these writes.